src/applications/mhv/medical-records/actions/vitals.js

```diff
@@ -12,10 +12,13 @@ export const getVitals = () => async dispatch => {
   }
 };
 
-export const getVitalDetails = vitalType => async dispatch => {
+export const getVitalDetails = (vitalType, vitalList) => async dispatch => {
   try {
-    await dispatch(getVitals());
-    dispatch({ type: Actions.Vitals.GET, vitalType });
+    if (vitalList) {
+      dispatch({ type: Actions.Vitals.GET, vitalType });
+    } else {
+      await dispatch(getVitals());
+    }
```

Comment:
There are a few problems here. `if(vitalList)` will not behave in the intended way if vitalList is empty or not a list. I think we have a helper function for that.

Also, `dispatch({ type: Actions.Vitals.GET, vitalType });` has to happen in either case in order to update Redux.